hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DataSourceOptions.scala

```diff
@@ -788,11 +788,11 @@ object DataSourceOptionsHelper {
   def inferKeyGenClazz(props: TypedProperties): String = {
     val partitionFields = props.getString(DataSourceWriteOptions.PARTITIONPATH_FIELD.key(), null)
     val recordsKeyFields = props.getString(DataSourceWriteOptions.RECORDKEY_FIELD.key(), DataSourceWriteOptions.RECORDKEY_FIELD.defaultValue())
-    genKeyGenerator(recordsKeyFields, partitionFields)
+    inferKeyGenClazz(recordsKeyFields, partitionFields)
   }
 
-  def genKeyGenerator(recordsKeyFields: String, partitionFields: String): String = {
-    if (partitionFields != null) {
+  def inferKeyGenClazz(recordsKeyFields: String, partitionFields: String): String = {
+    if (!StringUtils.isNullOrEmpty(partitionFields)) {
```

Comment:
Hi @danny0405 , seems we should judge empty "" for this, else we cannot get `NonpartitionedKeyGenerator` in any situation.

Reply:
Sure, we need to do that.